atc/api/jobs_test.go

```diff
@@ -577,6 +578,57 @@ var _ = Describe("Jobs API", func() {
 				Expect(response.Header.Get("Expires")).To(Equal("0"))
 			})
 
+			Context("when generates bagde title", func() {
+				It("uses url `title` parameter", func() {
+					response, err := client.Get(server.URL + "/api/v1/teams/some-team/pipelines/some-pipeline/jobs/some-job/badge?title=cov")
+					Expect(err).NotTo(HaveOccurred())
+
+					body, err := ioutil.ReadAll(response.Body)
+					Expect(err).NotTo(HaveOccurred())
+
+					Expect(strings.Contains(string(body), `
+      <text x="18.5" y="15" fill="#010101" fill-opacity=".3">cov</text>
+      <text x="18.5" y="14">cov</text>
+`)).Should(BeTrue())
+				})
+				It("uses default `build` title if not specified", func() {
+					response, err := client.Get(server.URL + "/api/v1/teams/some-team/pipelines/some-pipeline/jobs/some-job/badge")
+					Expect(err).NotTo(HaveOccurred())
+
+					body, err := ioutil.ReadAll(response.Body)
+					Expect(err).NotTo(HaveOccurred())
+
+					Expect(strings.Contains(string(body), `
+      <text x="18.5" y="15" fill="#010101" fill-opacity=".3">build</text>
+      <text x="18.5" y="14">build</text>
+`)).Should(BeTrue())
+				})
+				It("uses default `build` title if url `title` parameter is falsy", func() {
+					response, err := client.Get(server.URL + "/api/v1/teams/some-team/pipelines/some-pipeline/jobs/some-job/badge?title=")
+					Expect(err).NotTo(HaveOccurred())
+
+					body, err := ioutil.ReadAll(response.Body)
+					Expect(err).NotTo(HaveOccurred())
+
+					Expect(strings.Contains(string(body), `
+      <text x="18.5" y="15" fill="#010101" fill-opacity=".3">build</text>
+      <text x="18.5" y="14">build</text>
+`)).Should(BeTrue())
+				})
+				It("html escapes title", func() {
+					response, err := client.Get(server.URL + "/api/v1/teams/some-team/pipelines/some-pipeline/jobs/some-job/badge?title=%24cov")
+					Expect(err).NotTo(HaveOccurred())
+
+					body, err := ioutil.ReadAll(response.Body)
+					Expect(err).NotTo(HaveOccurred())
+
+					Expect(strings.Contains(string(body), `
+      <text x="18.5" y="15" fill="#010101" fill-opacity=".3">$cov</text>
```

Comment:
`$cov` isn't HTML escaped; should the example input be something like `<cov>` instead so it shows the escaping?

Reply:
Probably the test name isn't clear, because my intention was to verify that colling the endpoint with someting like  "badge?title=**%24cov**" resulted i "$cov" into the badge